### DISCUSSION TITLE
How to know if a file is downloaded? Can headers be used? in case you need selinum 4 how to configure it

### DISCUSSION THREAD

**Discussion by emontero-Nexplore** [webdriverio/webdriverio#8004] [Q&A]
How to know if a file is downloaded? Can headers be used? in case you need selinum 4 how to configure it

**Comment by christian-bromann** [comment]
> How to know if a file is downloaded?

Check if the UI shows a banner that the download has finished. You are testing from a user perspective. If the user doesn't see such a banner they will never know either.

**Reply by emontero-Nexplore** [reply]
I do that check, but I also want to check that the file was downloaded. I think the test will improve a lot

**Reply by christian-bromann** [reply]
You can pass in a download directory as argument to the browser (see chrome capabilities and chrome flags) and then check if the file is available. Note that this only works locally.

**Reply by emontero-Nexplore** [reply]
So that doesn't work for me either, although I knew how to do it, thanks anyway
